spec/teacher_spec.rb

```diff
@@ -0,0 +1,17 @@
+require_relative './spec_helper'
+
+describe Teacher do
+  teacher = Teacher.new(25, 'Math', 'Camila')
+
+  it 'should have an age equals to 25' do
+    expect(teacher.age).to eq(25)
+  end
+
+  it 'should return true if the teacher can use the services' do
+    expect(teacher.can_use_services?).to be true
+  end
+
+  it 'should have a specialization in Math' do
+    expect(teacher.specialization).to eq('Math')
+  end
+end
```

Comment:
- [ ] Your teacher object cannot be created due to errors. In the Teacher constructor, you need to assign `parent_permission` correctly by changing your `initialize` method to this

  ```
  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
    super(age, name, parent_permission: parent_permission)
    @specialization = specialization
  end
  ```

- [ ] After fixing that the test on line 15 will fail. You need to set the getter for `specialization` at the top of the class.

  ```
  attr_reader :specialization
  ```